staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -903,6 +908,7 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope RequestSco
 		if allowsOptions {
 			body, err := readBody(req)
 			if err != nil {
+				err = errors.NewBadRequest(err.Error())
```

Comment:
I'm not convinced we can claim any failure to read the body is a result of a malformed request. If we cannot then we shouldn't be blanket returning a 400 here.